packages/web3-eth-abi/src/coders/base/number.ts

```diff
@@ -0,0 +1,176 @@
+		// eslint-disable-next-line no-bitwise
+		const byteValue = Number(a & BigInt(0xff));
+		uint8Array[i] = byteValue;
+		// eslint-disable-next-line no-bitwise
+		a >>= BigInt(8);
+	}
+
+	// If the value is negative, convert it to two's complement.
+	if (isNegative) {
+		for (let i = 0; i < byteLength; i += 1) {
+			// eslint-disable-next-line no-bitwise
+			uint8Array[i] = ~uint8Array[i];
+		}
+		for (let i = byteLength - 1; i >= 0; i -= 1) {
+			const byteValue = uint8Array[i] + 1;
+			// eslint-disable-next-line no-bitwise
+			uint8Array[i] = byteValue & 0xff;
+			if (byteValue <= 0xff) {
+				break;
+			}
+		}
+	}
+
+	return uint8Array;
+}
+
+function uint8ArrayToBigInt(value: Uint8Array): bigint {
+	// eslint-disable-next-line no-bitwise
+	const isNegative = (value[0] & 0x80) !== 0; // Check the most significant bit for negativity
+
+	let result = BigInt(0);
+
+	// Convert the Uint8Array to a BigInt
+	for (const byte of value) {
+		// eslint-disable-next-line no-bitwise
+		result <<= BigInt(8);
+		// eslint-disable-next-line no-bitwise
+		result |= BigInt(byte);
+	}
+	// If negative, apply two's complement
+	if (isNegative) {
+		// eslint-disable-next-line no-bitwise
+		const mask = (BigInt(1) << (BigInt(8) * BigInt(value.length))) - BigInt(1);
+		// eslint-disable-next-line no-bitwise
+		result = -((~result + BigInt(1)) & mask);
+	}
+
+	return result;
+}
+
+const numberLimits = new Map<string, { min: bigint; max: bigint }>();
+
+// precalculate all the limits
+for (let i = 8; i <= 256; i += 8) {
+	numberLimits.set(`uint${i}`, {
+		min: BigInt(0),
+		max: BigInt(2) ** BigInt(i) - BigInt(1),
+	});
+	numberLimits.set(`int${i}`, {
+		min: -(BigInt(2) ** BigInt(i - 1)),
+		max: BigInt(2) ** BigInt(i - 1) - BigInt(1),
+	});
+}
+
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+numberLimits.set(`int`, numberLimits.get('int256')!);
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+numberLimits.set(`uint`, numberLimits.get('uint256')!);
+
+export function encodeNumber(param: AbiParameter, input: unknown): EncoderResult {
+	let value;
+	try {
+		value = toBigInt(input);
+	} catch (e) {
+		throw new AbiError('provided input is not number value', {
+			type: param.type,
+			value: input,
+			name: param.name,
+		});
+	}
+	const limit = numberLimits.get(param.type);
+	if (!limit) {
+		throw new AbiError('provided abi contains invalid number datatype', { type: param.type });
+	}
+	if (value < limit.min) {
+		throw new AbiError('provided input is less then minimum for given type', {
+			type: param.type,
+			value: input,
+			name: param.name,
+			minimum: limit.min.toString(),
+		});
+	}
+	if (value > limit.max) {
+		throw new AbiError('provided input is greater then maximum for given type', {
+			type: param.type,
+			value: input,
+			name: param.name,
+			maximum: limit.max.toString(),
+		});
+	}
+	return {
+		dynamic: false,
+		encoded: bigIntToUint8Array(value),
```

Comment:
It works well but a bit confusing at first look, as its padded uint8array , can rename this function? to encoded or padded suffix? so its not misused else where.

Reply:
It's not exported, just in this file.